### ISSUE TITLE
Five shapes of function name are read as no name, so `function-whitespace-after` takes their parentheses for a group's

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#252]
Found while fixing [#230](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/230), which teaches `function-whitespace-after` to read only the parenthesis a call closes, by way of `findFunctionArgumentSpans`. That utility names a call by what stands in front of its parenthesis, and reads a name with `TRAILING_FUNCTION_NAME`. The pattern reads a run of ASCII letters, digits, underscores and hyphens, and a CSS identifier is both wider than that and not a run of characters at all, so five shapes of name are read as no name and their parentheses are taken for a group's.

### What minimal example or steps are needed to reproduce the bug?

Five calls, each named by something the pattern cannot read. The first four are reported only where they stand among the arguments of another call, and the fifth wherever it stands:

```less
a { b: translate(%("%dpx", @a)red); }
```

```css
a { b: translate(日本(1)red); }
```

```css
a { b: translate(foo-(1)red); }
```

```css
a { b: translate(--1(2)red); }
```

```css
a { b: fo\6f(1)red; }
```

The first is Less's format function, whose name is one character of punctuation. The second is a name written outside ASCII, which the CSS grammar allows anywhere an identifier stands: every code point at U+0080 and above is an identifier code point. The third is a name closing on a hyphen, which the grammar allows as well: an identifier may open with a hyphen and may close with one, and only what stands *after* a leading hyphen is restricted. The fourth is a name opening on `--` and a digit, which the grammar allows because the two hyphens are enough to start an identifier by themselves. The fifth is a name spelling one of its characters as an escape, `fo\6f` being `foo`: the run of characters in front of the parenthesis reads `6f`, which opens on a digit and so names nothing.

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/function-whitespace-after": "always"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

`Expected whitespace after ")"` in all three, and the fix writing the space in. Each of the three is a call, and the word behind it is a value of its own.

### What actually happened?

Nothing reported, and nothing written. Until [#230](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/230) the warning was there, so these are false negatives that branch left behind, named in its pull request rather than passed over in silence.

The same three calls standing at the head of a value drew nothing before that branch either: `style-search`, which the rule used to place its matches with, asks for an ASCII letter in front of a parenthesis and finds none in front of any of the three. So the reading was always narrow at the head of a value, and what covered the nested shape was the accident that `style-search` reads every parenthesis standing inside a call's arguments, whatever named it.

### What is the cause?

`TRAILING_FUNCTION_NAME` is `/(?:(?<![\w-])-*[A-Za-z_](?:[\w-]*\w)?)?$/u`, and `\w` is ASCII whatever flags it carries. A CSS identifier is wider in three directions: it may hold code points outside ASCII, it may close on a hyphen, and behind a leading `--` it may open on anything at all. Less adds one name of its own outside the grammar, `%`. And an identifier is not a run of characters: an escape spells one character with several, so no pattern reading a trailing run can tell `fo\6f(` from a run opening on a digit.

### Do you have a proposal to fix the bug?

Read a CSS identifier: the non-ASCII code points alongside the ASCII ones, a hyphen at the close as well as at the head, and anything behind a leading `--`. Escapes are the half of this that a wider character class cannot answer. Where the escape opens the name, `\66 oo(1)`, the run behind it reads `oo` and the call is found whichever pattern reads it; where it stands inside the name, `fo\6f(1)`, the run reads `6f` and the call is lost; and where the name is nothing but an escape, `\-(1)` and `\31 23(1)`, nothing is found either, though the pattern as it stood before did find one in each. Reading them properly means reading the name forwards, the way the scan of the utility already steps over a backslash, rather than backwards from the parenthesis.

The pattern is shared, so whatever is done to it answers for `mediaFeatureColonSpaceChecker`, `findMediaOperator` and `mediaQueryListCommaWhitespaceChecker` as well as for `function-whitespace-after`, and there it costs the other way about: a call the pattern cannot name has its arguments read as media syntax, so `@media a-(max-width:600px)` and `@media fo\6f(max-width:2px)` are reported by the eight rules of those three utilities and a fix written there lands inside the call. Widening the name closes that at the same time, and no media feature is spelled `%(…)` or named outside ASCII in any file anyone has reported.

Note what must not be widened with it, and note that a name closing on a hyphen is the delicate one of the five. A lone hyphen and a hyphen behind a number name nothing, and reading them as names is what [#230](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/230) had to close: `h1 { width: -(@a * 2)px; }` and `h1 { width: 2-(1)px; }` are a negated and a subtracted calculation, and a call read there is the corruption that issue is about. So `foo-(1)` has to be told from `2-(1)` and from `-(1)`, which the run's opening character already does; the whole run must be read, never a suffix of it, or `2-` becomes the name `-` again.

One refusal is not a shape of this and must survive the fix: a run opening on a digit names nothing, `2px(` and `2and(` being a dimension in front of a parenthesis rather than a call. [#230](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/230) stopped reading those, and reading them again would be a step back.
